code/client/munkilib/updatecheck/core.py

```diff
@@ -99,9 +99,9 @@ def check(client_id='', localmanifestpath=None):
         # stop precaching_agent if it's running
         download.stop_precaching_agent()
 
-        # prevent idle sleep only if we are on AC power
+        # prevent idle sleep only if we are on AC power or SuspendDisplaySleepDuringBootstrap set
         _caffeinator = None
-        if powermgr.onACPower():
+        if powermgr.onACPower() or prefs.pref('SuspendDisplaySleepDuringBootstrap'):
```

Comment:
The preference is named "SuspendDisplaySleepDuringBootstrap" and yet we're not checking if we are bootstrapping. So an admin sets this, and now users have their laptops run out of batter because Munki keeps suspending sleep when it runs.
I already am skeptical about the wisdom of blocking sleep when we are not on AC power, but this just makes it worse.

Reply:
Yea I totally agree, I thought about breaking that out as another preference `SuspendSleepOnBattery`, and using that here. I did notice that just the update check gates sleep to only on AC power, while an install will block sleep either way.